pandas/tseries/converter.py

```diff
@@ -172,7 +172,7 @@ def _dt_to_float_ordinal(dt):
     is a :func:`float`.
     """
     if (isinstance(dt, (np.ndarray, Index, Series)
-                   ) and is_datetime64_ns_dtype(dt)):
```

Comment:
Theoretically, this is not fully correct, as `is_datetime64_any_dtype` can also accept datetime64 values with other than ns resolution? 
Although I don't think you can end up here with such values when going through the DatetimeConverter, as datetime64 arrays first get passed through `to_datetime`